server/util/networking/networking.go

```diff
@@ -236,9 +240,12 @@ func (n *HostNet) Unlock() {
 }
 
 // Get assigns a host network IP for the given VM index.
-func (a *HostNetAllocator) Get(vmIdx int) (*HostNet, error) {
+func (a *HostNetAllocator) Get() (*HostNet, error) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
 	for attempt := 0; attempt < numAssignableNetworks; attempt++ {
-		netIdx := (vmIdx + attempt) % numAssignableNetworks
+		netIdx := a.idx % numAssignableNetworks
```

Comment:
nit: don't need to % numAssignableNetworks if you're doing it in the line below

Reply:
good catch, removed